### PR TITLE
add tests for bottom step button

### LINKED ISSUE #1369: Writing UI test and making UI more stable
With #1366 we were able to set up a basic browser-based UI test and it works flawlessly. So lets build more tests and make the UI stable.
Test yet to be built:

- [ ] Clicking add step adds a new box and a different image

- [ ] Clicking clear step button clear all the step

- [ ] Clicking Delete Step removes the image

- [ ] Select a module to have all the modules with the correct name(we faced issues when modules have which is undefined in the past ).

- [ ] Insert step works as expected.

- [ ] Input types(range,text etc) of all the modules are correct-
we need to fix this before we can write test on it

### PR DESCRIPTION
**Pull request by pythongiant**
concerns #1369  (<=== Replace `0000` with the Issue Number)

Hi, I have completed task 1 specified in #1369 

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

### CODE REVIEW

**Comment by pythongiant** [comment]
image
Huh Travis Failed but locally my tests worked

**Comment by pythongiant** [comment]
@keshav234156 @debck @ananyaarun

**Comment by pythongiant** [comment]
CC: @jywarren

**Comment by pythongiant** [comment]
Works now

File: test/ui-2/test/DropDownButtonStep.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const Length = await page.evaluate(() => document.querySelectorAll('.step').length);
+    await page.click('[data-value=\'brightness\']');
+
+    await page.waitForSelector('#add-step-btn');
+    page.click('#add-step-btn');
+    const LengthChanged = await page.evaluate(() => document.querySelectorAll('.step').length);
+    expect(Length).toBe(1);
+    expect(LengthChanged).toBe(2);
+
+  });
```
Comment (debck) [inline test/ui-2/test/DropDownButtonStep.test.js]: I think you forgot to put the `timeout` here as the async function ends.

**Comment by debck** [comment]
Also, change in the PR description from Fixes to `concerns` so that merging this doesn't close the issue.

File: test/ui-2/test/DropDownButtonStep.test.js (hunk 1 above)
Comment (pythongiant) [reply test/ui-2/test/DropDownButtonStep.test.js]: Done @debck

**Comment by keshav234156** [comment]
@pythongiant This is testing radio button not the add step.For that choose some different value which is not present in radio group

File: test/ui-2/test/DropDownButtonStep.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const fs = require('fs');
+beforeAll(async () => {
+  path = fs.realpathSync('file://../examples/index.html');
+  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+});
+describe('add step dropdown', ()=>{
+  test('Click Button', async()=>{
+    await page.waitForSelector('.step');
+    const Length = await page.evaluate(() => document.querySelectorAll('.step').length);
+    await page.click('[data-value=\'brightness\']');
```
Comment (keshav234156) [inline test/ui-2/test/DropDownButtonStep.test.js]: Choose different value not present in radio group

File: test/ui-2/test/DropDownButtonStep.test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    await page.waitForSelector('#add-step-btn');
+    page.click('#add-step-btn');
```
Comment (keshav234156) [inline test/ui-2/test/DropDownButtonStep.test.js]: add await

File: test/ui-2/test/DropDownButtonStep.test.js (hunk 2 above)
Comment (keshav234156) [reply test/ui-2/test/DropDownButtonStep.test.js]: you also need to add $(".selectize-input").click() before this.

Comment (pythongiant) [reply test/ui-2/test/DropDownButtonStep.test.js]: Umm @keshav234156 why are we choosing value not present there

**Comment by keshav234156** [comment]
@pythongiant look good now you have check src of the first image is different from the second image so that it assures us that different step is being added. You can do it with 
 expect($('.step:last img')[0].src).not.toEqual($('.step:first img')[0].src);
with small changes as required .

This won't work for the blend as steps have to be greater than 2 otherwise src of both would be equal. So change to some other module like blur.

**Comment by pythongiant** [comment]
@keshav234156 i have done the required changes. Check it out now?

**Comment by keshav234156** [comment]
@pythongiant Great !!!you finally did it.

**Comment by pythongiant** [comment]
@debck accept and merge?

**Comment by pythongiant** [comment]
@jywarren merge?

**Comment by pythongiant** [comment]
@HarshKhandeparkar I have refactored the code a bit, check it out?

File: test/ui-2/test/DropDownButtonStep.test.js [hunk 4]
```diff
@@ -6,18 +6,28 @@ beforeAll(async () => {
...
+    // click dropdown and select appropriate option. then submit
     await page.click('.selectize-input');
     await page.click('[data-value=\'blend\']');
     await page.waitForSelector('#add-step-btn');
     await page.click('#add-step-btn');
+ 
+    // Get Changed Length
     const LengthChanged = await page.evaluate(() => document.querySelectorAll('.step').length);
+
+    //Check If Image Of Step actually Changed
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/DropDownButtonStep.test.js]: Please add a space in the comment

File: test/ui-2/test/DropDownButtonStep.test.js [hunk 5]
```diff
@@ -6,18 +6,28 @@ beforeAll(async () => {
...
 describe('add step dropdown', ()=>{
   test('Click Button', async()=>{
+    // Wait for .step to load
     await page.waitForSelector('.step');
+    /// Get Length Of steps
     const Length = await page.evaluate(() => document.querySelectorAll('.step').length);
+
+    // Wait for .selectize-input to load
     await page.waitForSelector('.selectize-input');
+    // click dropdown and select appropriate option. then submit
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/DropDownButtonStep.test.js]: Please start with a capital letter.

**Comment by keshav234156** [comment]
@pythongiant Please do the changes as requested by @HarshKhandeparkar

**Comment by pythongiant** [comment]
@keshav234156 check it out?

File: test/ui-2/test/DropDownButtonStep.test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+const timeout = process.env.SLOWMO ? 30000 : 10000;
+const fs = require('fs');
+beforeAll(async () => {
+  path = fs.realpathSync('file://../examples/index.html');
+  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+});
+describe('add step dropdown', ()=>{
```
Comment (HarshKhandeparkar) [inline test/ui-2/test/DropDownButtonStep.test.js]: Let's make this look professional. Could you please capitalize the test name too?

**Comment by pythongiant** [comment]
@HarshKhandeparkar fixed it!

**Comment by jywarren** [comment]
Fantastic!!

**Pull request merged**